Implement dvc.org redesign to include dvcx coming soon

yathomasi opened the pull request:
- Closes https://github.com/iterative/dvc.org/issues/4705
- Preview URL: https://dvc-org-dvcx-uc2wlxyiggynway5p.herokuapp.com

Remaining Checklist:

- [x] Action for `Get on the waitlist` CTA
- [x] DVC<sup>X</sup> CLI commands

yathomasi commented:
I think we also need to update this visualization to match the gradient. cc: @djsauble 

<img width="1254" alt="Screenshot 2023-08-01 at 18 01 55" src="https://github.com/iterative/dvc.org/assets/20840228/e2a242c4-dd35-4e88-9713-d59e702f862d">

djsauble commented on src/components/Home/Hero/HeroTitleSection.tsx:
```diff
@@ -0,0 +1,55 @@
+            'lg:mr-16'
+          )}
+        >
+          <h1 className={cn('leading-10 font-extralight', styles.title)}>
```
These lines look a little squashed. Can you increase the line-height of the `h1` to 1.5 and add a 16px margin between the `h1` and the `p` which follows?

djsauble commented on src/components/Home/Hero/HeroTitleSection.tsx:
```diff
@@ -0,0 +1,55 @@
+    <div className={className}>
+      <div
+        className={cn(
+          'text-center ',
```
I'd like the columns to be vertically centered so that the white space in the hero feels more balanced. Something like this maybe?

```suggestion
          'text-center items-center',
```

djsauble commented on src/components/Home/Hero/GetStarted/GetStartedWithDvcX.tsx:
```diff
@@ -0,0 +1,32 @@
+          alt="DVCx Logo"
+        />
+      </div>
+      <DvcxSlides />
```
Can you reverse the order of the terminal and accordion for DVCX? Accordion on the right, terminal on the left.

This will help increase the visual separation between the two "Get Started" sections and help people understand how this section aligns with the user journey diagram from the hero above.

djsauble commented on src/components/Home/Hero/GetStarted/GetStartedWithDvc.tsx:
```diff
@@ -0,0 +1,93 @@
+          alt="DVC Logo"
+        />
+      </div>
+      <DvcSlides />
```
Can you change the background color of this terminal to #EEF4F8 to match the DVC half of the hero above?

This helps create a visual connection so that people intuitively understand where this section fits into the overall user journey.

djsauble commented on src/components/Home/index.tsx:
```diff
@@ -21,11 +22,11 @@ const Home: React.FC = () => {
-      <HeroSection className={styles.heroSection}>
-        <LandingHero />
+      <Hero />
+      <HeroContainer className={styles.heroContainer}>
```
The company logos section should have a background color of #EEF4F8 to provide more visual separation from the Get Started with DVC section above.

djsauble commented on src/components/Home/Hero/HeroSection.tsx:
```diff
@@ -0,0 +1,157 @@
+            >
+              <Badge className="bg-dark text-light">
+                {stars && shortenNumber(stars, 1)}
+                <StarGithub />
```
The star doesn't show up in the preview URL you shared. There's just an empty space where the star should be.

![image](https://github.com/iterative/dvc.org/assets/186715/141ea335-9e19-4c2b-b6a9-7cbf9645a93c)

djsauble commented on src/components/Home/Hero/GetStarted/GetStartedWithDvc.tsx:
```diff
@@ -0,0 +1,93 @@
+    <HeroContainer className="py-10 px-6" id="get-started-dvc">
+      <div className="flex items-center gap-4 md:justify-center">
+        <h1 className={cn('text-2xl font-medium')}>Get Started with</h1>
+        <StaticImage
```
Small thing, but I'd like for the DVC and DVCX logos to feel like they're part of the `h1` here. For example, the gap between "Get started with" and the logo should be about the same as a whitespace character, and the baseline of the 'D' and 'C' in the logo should be aligned with the baseline of the "Get started with" section.

I realize this is probably fiddly, so don't worry too much about it, but I think it'll look nicer if we get these visual details honed in. 

![image](https://github.com/iterative/dvc.org/assets/186715/531d9480-5285-4a7e-9850-5cf7b0a40d4e)

djsauble reviewed (changes requested):
Thanks for all of your work on this @yathomasi. It's definitely getting there. 👍

yathomasi commented:
@djsauble I have addressed most of the comments. Feel free to go through them once again.

djsauble commented:
Looking good! Just a couple other things I'm noticing.

The centered elements on each side of the DVCX/DVC hero don't seem to respect the max width of the page. On wide screens they start to drift off to the left and right, respectively (see screenshot below).

![Screenshot 2023-08-02 at 4 10 53 PM](https://github.com/iterative/dvc.org/assets/186715/11e6453b-0d74-4712-a9e4-2323c49f6f39)

Also, I'm still not seeing the GitHub star icon.

![Screenshot 2023-08-02 at 4 09 07 PM](https://github.com/iterative/dvc.org/assets/186715/4b8574a4-e8c1-4ce9-af7e-6dfe09b1e118)

yathomasi commented:
Thanks, @djsauble.

> The centered elements on each side of the DVCX/DVC hero don't seem to respect the max width of the page. On wide screens they continue to drift off to the left and right,

I have updated it accordingly.

> Also, I'm still not seeing the GitHub star icon.

Okay, I hadn't checked with Safari. It's fixed now.


> Remaining Checklist:
>
> * [ ]  DVCX CLI commands

cc: @volkfox 

> Remaining Checklist:
> 
> * [ ]  Action for `Get on the waitlist` CTA

@djsauble How do you think we want to get the email here? Do you have any idea?

It could be modal if there are multiple fields else it can just be small popup if it's just email we want or we can simply have filed similar to subscribe section.

Pinging @jurv11 for necessary form to be in place.

djsauble commented:
Looks good!

Let's try adding the email input field inline (similar to the subscribe section).

yathomasi commented:
> Let's try adding the email input field inline (similar to the subscribe section).

Here is how it looks.  Let me know what you think.

<img width="1728" alt="Screenshot 2023-08-04 at 01 08 32" src="https://github.com/iterative/dvc.org/assets/20840228/f675ab7b-767a-45c1-985a-d741aaade027">


> since this form will not be a lead generator for sales, he will not be involved in setting it up. 

👍 

> I'm not sure where the emails from the subscribe section go, but this waitlist form should behave similarly.

We have them in the Mailchimp. I can create a separate audience list for this. I think @jendefig will also be involved here as I think we will also need to have some emails templates ready for 
1. Thank you for joining the waitlist
2. Release announcements (This I think will be together with all other subscribers)
3. Maybe other DVCx-related updates

yathomasi commented:
Here, the `Join waitlist` will add to the DVCx audience in Mailchimp.

<details>
<summary>Current Demo</summary>

https://github.com/iterative/dvc.org/assets/20840228/77d3321d-1f6b-43b7-a69c-eb7188046998

</details>

I wanted this to simply work without redirect but unfortunately, it only worked in the local server but not in the live. 
I did tried bunch of method eg, submitting inside a function (failed with CORS), tried setting up backend API but worked locally but didn't worked on server, also tried using `mailchimp` API SDK, didn't worked either.

<details>
<summary>Worked only locally</summary>

https://github.com/iterative/dvc.org/assets/20840228/a0ce2431-0d86-4e3a-8a2f-0cd2dc757864

</details>

Anyway, moved on with a setup similar to the previous one with redirection. 

Now, we need to update the `DVCx` command line to merge this. 
Could you please help out with that? @djsauble @volkfox @dmpetrov @shcheklein

djsauble commented:
> Now, we need to update the `DVCx` command line to merge this. Could you please help out with that? @djsauble @volkfox @dmpetrov @shcheklein

@volkfox and @dmpetrov will be helping with this.

omesser commented on src/components/Home/Hero/HeroSection.tsx:
```diff
@@ -0,0 +1,189 @@
+
+  return (
+    <>
+      {/* Title Section */}
```
@djsauble @yathomasi  - I have some HL feedback about the design of the hero section (black/white halves)


Even though I already saw the design before and knew what to expect, when I first loaded the page it "looked broken", why?

- the hero section is cut even on larger screens until you scroll to it
- no horizontal borders

This means my mind initially "ignored" the white DVC half (right) and it looked as if the page was the left DVCx part, with black background, and it was somehow offset and overflowed to the left.

I understand the original design but this first impression is a bit jarring.

Maybe allowing borders (bg gradient-colour) on the sides (not letting the white/black section fill the entire width) would help center the eyes?

wdyt @djsauble ?

<details>
  <summary>this is how it looked like on a 24'' screen</summary>
  
  ![Screenshot 2023-08-18 at 11 58 59](https://github.com/iterative/dvc.org/assets/18181802/9de35641-2974-4c73-a9a5-cb582aaa003b)

</details>

djsauble replied:
I agree with you. We're trying to not change the design of the page more than necessary, but I think this would benefit from some tweaks.

This is what it looks like with a max-width and even a drop-shadow to add some continuity to the two sides of the section. wdyt? cc @yathomasi 

<img width="1848" alt="Proposed design (Desktop)" src="https://github.com/iterative/dvc.org/assets/186715/141ba136-c81e-4a6b-b0b7-f6c7833a912c">

omesser replied:
Thanks @djsauble  - it looks better to me personally, eliminating the effect which bothered me a bit!

djsauble replied:
On discussing this with @volkfox, since the new dvc.ai splash screen uses a similar black/white background motif, we want to avoid creating the same effect on this page. Instead, we'll remove the solid background color behind the diagram.

https://www.figma.com/file/nq7SOejyDtiaoxk4FplvyO/DQL?type=design&node-id=2486%3A2515&mode=design&t=ymxljJn77adO69ER-1

djsauble replied:
https://github.com/iterative/dvc.org/pull/4803

omesser reviewed (changes requested):
Nicely done @yathomasi ! 🤩 


left a few comments, but looks good!

djsauble commented on src/components/Home/Hero/HeroSection.tsx:
```diff
@@ -0,0 +1,189 @@
+          <div
+            className={cn('flex flex-col gap-4', 'lg:flex-row lg:items-center')}
+          >
+            <h2 className="text-3xl font-medium">Dataset versioning</h2>
```
@dmpetrov @volkfox ^

djsauble replied:
@volkfox has decided to rename this to "Dataset factory" and the DVC side to "Data and model versioning"

djsauble replied:
https://github.com/iterative/dvc.org/pull/4801

djsauble commented on src/components/Home/Hero/GetOnTheWaitlist/Form.tsx:
```diff
@@ -0,0 +1,112 @@
+          name="subscribe"
+          id="mc-embedded-subscribe"
+        >
+          {loading && <Spinner className="mr-3" />} Get on the waitlist
```
@dmpetrov @volkfox Are we planning to synchronize the launch of the dvc.org redesign with the launch of the DVCX CLI? If so, I agree with @omesser that we could just turn this into a download button.

djsauble commented on src/components/Home/Hero/HeroTitleSection.tsx:
```diff
@@ -0,0 +1,48 @@
+            'lg:pr-16'
+          )}
+        >
+          <h1 className={cn('font-extralight', styles.title)}>
```
Both DVCX and DVC are technically free and open source (via their CLIs), so I'm not sure that swapping the order will increase the clarity of the design at all.

omesser replied:
DVCx CLI is still not open sourced (will be soon I hope? but don't know when).
Also, DVCx is enterprise and commercially oriented by its value proposition (it's not open-source first, and the value prop of it's open source CLI only flavor isn't as strong).
Additionally, DVCx will not have the open source community (and thus association/impact) as DVC any time soon I guess - DVC got 7y of mileage.
So out of the 2, to my understanding, there's a clear association of: DVC -> OSS, DVCx -> enterprise first product (commercial)

yathomasi commented:
Thanks, @omesser & @djsauble  for the feedback/suggestion. I have updated the Hero Visualization section to make it isolated on a large screen and also updated the typo to `Join the waitlist`.

omesser commented:
@yathomasi 
Any idea why I'm seeing this on the PR deployment app:

<details>
  <summary> expand for screenshot </summary>
  <img width="1728" alt="Screenshot 2023-08-23 at 17 14 29" src="https://github.com/iterative/dvc.org/assets/18181802/4c1e9f46-bbad-48ef-aeeb-aa33ce9e2b89">

</details>

yathomasi commented:
> Any idea why I'm seeing assets missing (images, low fidelity icons) this on the PR deployment app?

TBH, I am not sure what was the reason as it was all good on my testing on local deployment. I can guess it would be related to Heroku not able to serve the files properly. 

Anyway, I have redeployed and it seems to be working fine. Let me know if it's not, I can deep dive. cc: @omesser

djsauble commented on src/components/Home/Hero/HeroSection.tsx:
```diff
@@ -0,0 +1,199 @@
+          <div
+            className={cn('flex flex-col gap-4', 'lg:flex-row lg:items-center')}
+          >
+            <h2 className="text-3xl font-medium">ML model versioning</h2>
```
```suggestion
            <h2 className="text-3xl font-medium">Data and model versioning</h2>
```

djsauble replied:
https://github.com/iterative/dvc.org/pull/4801

djsauble commented on src/components/Home/Hero/HeroTitleSection.tsx:
```diff
@@ -0,0 +1,48 @@
+        >
+          <h1 className={cn('font-extralight', styles.title)}>
+            <span className="font-medium block sm:inline md:block">
+              Data Version Control
```
@volkfox has requested that we add "and much more" to this to pay homage to the brand name while emphasizing that DVC has moved beyond mere data version control.

Something like this?

```suggestion
              Data Version Control<br/><em>&mdash; and much more &mdash;</em>
```

djsauble replied:
https://github.com/iterative/dvc.org/pull/4799

djsauble commented on src/components/Home/Hero/BetterTogether/index.tsx:
```diff
@@ -0,0 +1,20 @@
+
+const BetterTogether = () => {
+  return (
+    <HeroContainer className="py-14 bg-gray-hover text-light">
```
Let's change this to dark text on light background so that the separation between the "Get Started with DVCX" and "Get Started with DVC" sections is not so stark.

This is probably wrong but something along these lines?

```suggestion
    <HeroContainer className="py-14 bg-gray-100">
```

djsauble replied:
https://github.com/iterative/dvc.org/pull/4800

djsauble commented on src/components/Home/Hero/HeroTitleSection.tsx:
```diff
@@ -0,0 +1,48 @@
+        </div>
+        <div className={cn('flex-1', 'md:text-justify', 'lg:pl-16')}>
+          <p className="text-xl font-extralight md:pl-8 lg:pl-0 leading-10">
+            Manage and version unstructured image, audio, video, and text files
```
```suggestion
            Manage and version images, audio, video, and text files
```

djsauble replied:
https://github.com/iterative/dvc.org/pull/4799

djsauble reviewed (changes requested):
Had a discussion with @volkfox yesterday on the design of this page, and I've added a few additional comments. Sorry about all the churn on this one!

djsauble commented:
One other thing that @volkfox brought up. We need a (subtle) link to cloud.dvc.ai from this page for people who come here and realize that they actually want to buy a product instead of using OSS.

To accommodate this, I've added a link to the immediate right of the "Get Started" button in the header that will send people to cloud.dvc.ai.

![Uploading Proposed design (Desktop).png…]()

jendefig commented:
Just doing the metrics today and wanted to point out that it's safe to say the Github Stars number can be changed to 12k + for the release. We are currently at 11,923 and gain over 100 each month.

yathomasi commented:
> Just doing the metrics today and wanted to point out that it's safe to say the GitHub Stars number can be changed to 12k + for the release. We are currently at 11,923 and gain over 100 each month.

I guess it's okay as it is as we fetch the number live and show them. Once we get there it will be reflected.

omesser commented on src/components/Home/Hero/HeroSection.tsx:
```diff
@@ -0,0 +1,191 @@
+            }}
+          >
+            <span>
+              Download DVC<sup>X</sup>
```
I would not label buttons that navigate/scroll "Download".
Maybe "Get DVC/DVCx" ?

omesser reviewed (approved):
🤩 Looks better IMO!
Left a comment ("fake" download buttons can be annoying)
approving anyways for my part

shcheklein commented:
@djsauble @volkfox @dmpetrov what is the plan for this folks?

djsauble commented:
The plan is to deploy this once DVCX release happens. People need to be able to download the CLI from this site.

We also still need real DVCX CLI commands added. cc @dmpetrov @volkfox

djsauble commented:
I chatted with @dmpetrov about this today. For DVCX commands, we'll use a query defined in a file:

**my-query.py**
```
from dql.query import C, DatasetQuery
from dql.lib.webdataset import WebDataset
from dql.lib.gpt4_vision import DescribeImage

prompt = "How many peope in the image? I need only number without any text."

people = Dataset("s3://my-storage/")                       \
              .filter(C.name.glob("*.jpg"))                \
              .map(DescribeImage(prompt, max_token = 300)) \
              .mutate(num_people = int(C.descr))

people.filter(C.num_people > 3)
```

…paired with a `dvcx query` command. e.g:

```

djsauble commented:
I chatted with @dmpetrov about this today, and the plan is to pair a query file with a CLI command. So:

**my-query.py**
```
from dql.query import C, DatasetQuery
from dql.lib.webdataset import WebDataset
from dql.lib.gpt4_vision import DescribeImage

prompt = "How many peope in the image? I need only number without any text."

people = Dataset("s3://my-storage/")                       \
              .filter(C.name.glob("*.jpg"))                \
              .map(DescribeImage(prompt, max_token = 300)) \
              .mutate(num_people = int(C.descr))

people.filter(C.num_people > 3)
```

....paired with...

**CLI**
```
$ dvcx query my-query.py my-dataset
.... progress
Dataset 'my-dataset' created
```

djsauble commented:
And since we haven't released DVCX, we'll replace the "Download" CTAs with "Coming soon" CTAs.

dmpetrov commented:
Let's deploy to prod now!

Some feedback. Not urgent, this can be implemented after the deployment.

- [ ] Remove CML from Other Tools (footer)
- [ ] CLI on DVC has a bit strange color - it's not easy to read
- [ ] Studio --> DVC Studio
- [ ] CLI on DVCX is too fast 😅  It should be 2x slower.
- [ ] Should we replace `by Iterative.ai` to `by dvc.ai`?

yathomasi commented:
> * CLI on DVC has a bit strange color - it's not easy to read

We no longer have the concept of white or dark, so I think it's best to use the dark theme for DVC commands as well, as the current cli theme was optimized for the dark theme and not the light theme. cc: @djsauble
